ipfx/dataset/ephys_data_set.py

```diff
@@ -33,57 +46,79 @@ class EphysDataSet(object):
     VOLTAGE_CLAMP = "VoltageClamp"
     CURRENT_CLAMP = "CurrentClamp"
 
-    def __init__(self,
-                 data: EphysDataInterface,
-                 sweep_info: Optional[Dict[str, Any]] = None):
-
-        self.data = data
-        self.ontology = data.ontology
-
-        if sweep_info is None:
-            sweep_info = self.extract_sweep_info()
+    @property
+    def ontology(self) -> StimulusOntology:
+        """The stimulus ontology maps short-form stimulus codes to complete
```

Comment:
Not exactly: it maps codified description of the stimulus type to the meaningful description.  The stimulus code may actually be longer than stimulus name